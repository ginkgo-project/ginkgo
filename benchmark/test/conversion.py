#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]', "-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# stdin
test_framework.compare_output(
    ["-formats", "coo,csr"],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    [
        "-input",
        str(test_framework.sourcepath / "input.mtx.json"),
        "-formats",
        "coo,csr",
    ],
    expected_stdout="conversion.simple.stdout",
    expected_stderr="conversion.simple.stderr",
)

# check that all conversions work
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr,ell,sellp,hybrid",
    ],
    expected_stdout="conversion.all.stdout",
    expected_stderr="conversion.all.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-formats",
        "coo,csr",
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="conversion.profile.stdout",
    expected_stderr="conversion.profile.stderr",
)
