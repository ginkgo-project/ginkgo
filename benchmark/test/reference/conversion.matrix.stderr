Running on reference(0)
Running with 2 warm iterations and 10 running iterations
The random seed for right hand sides is 42
The formats are coo,csr
Running test case <filename>
Matrix is of size (36, 36), 208
	Running conversion: coo-read
	Running conversion: coo-csr
	Running conversion: csr-read
	Running conversion: csr-coo
